### ISSUE TITLE
[package] protobuf/3.19.1: Protobuf_PROTOC_EXECUTABLE no longer set in conan 1.44

### ISSUE THREAD

**Issue by Wuestenschiff** [conan-io/conan-center-index#8625]
I am using a custom command to compile protobuf files, like this:

```cmake
add_custom_command(
    COMMENT "Generating protobuf files from proto definitions..."
    OUTPUT ${proto_srcs} ${proto_hdrs}
    COMMAND ${Protobuf_PROTOC_EXECUTABLE}
    ARGS --cpp_out ${CMAKE_CURRENT_LIST_DIR}/generated
		--proto_path=${CMAKE_CURRENT_LIST_DIR}
		--proto_path=${protobufffer_INCLUDE_DIRS}
		${protos}
    DEPENDS ${protos})
```
Since updating to 1.44 `${Protobuf_PROTOC_EXECUTABLE}` is no longer set. (No other changes). I tried to use `tool_requires` and `build_requires` but I did not manage to get it exported anymore. We us build and host profiles but even without them I was not able to get the variable exported.

**Comment by SpaceIm** [comment]
It's hard to answer without your conanfile, conan generators, conan command and CMakeLists.

**Comment by Wuestenschiff** [comment]
I can understand... but my project is not really a minimal working example. I can try to do one but this will need some time. I hopped this woud  ring some bells here. Not sure what has changed in 1.44 but ${Protobuf_PROTOC_EXECUTABLE} is no longer set. 1.43 works so I hopped his would be enough information. (Our cmake has mutiple k lines of code)

**Comment by Wuestenschiff** [comment]
@SpaceIm Have adjasted an old example so it runs in this problem it seems also to have something to do with the generator. cmake_find_package works but CMakeDeps does not.

This still build grpc 3.17.1 and not 3.19.1 but the problem is the same for both version and the recie did not change in those parts

[gRPC.zip](https://github.com/conan-io/conan-center-index/files/7816338/gRPC.zip)

It won't be able to do the custom command on the protofiles because ${Protobuf_PROTOC_EXECUTABLE} is not set.

Also works with 1.43 but not 1.44

**Comment by sburton84** [comment]
This sounds like the issue described in https://github.com/conan-io/conan/issues/10282 where @lasote has suggested that the issue is caused by the protobuf recipe setting `cmake_build_modules` on individual components, which it apparently should not be doing.

**Comment by Wuestenschiff** [comment]
@SpaceIm As I understand the other threads this seems to be a bug in the protobuf repo than. Have you guys plans to fix this or should we look into workarounds?

**Comment by SpaceIm** [comment]
I've tested https://github.com/conan-io/conan-center-index/pull/8820 with CMakeDeps and it seems to work.

**Comment by Wuestenschiff** [comment]
Nice I have subscribe to that pull request, and will test once it is merged. I am quite sure my POC works bur you need to remove the Cache because the protoc Variable is cached. But if there is development that is good enough for me. Sorry for being so impatient- but I had just fixed teh version number of conan to 1.43 for know and am a little anxious this will break soon one people start using the new 1.44 features.

**Comment by lasote** [comment]
I don't know for sure if the issue is related to that but we are releasing minor versions (`1.44.1` and `1.43.3`) with some bug fixes, including [this one](conan-io/conan#10282), probably today. Let me know if it works.

**Comment by jellespijker** [comment]
I believe I experience the same problems or related problems see conan-io/conan-center-index#8534 even after the merge of conan-io/conan-center-index#7857
